### PR TITLE
Enable MetricsAPMIT and change the assertion that logs on failure

### PR DESCRIPTION
**Pull request by pgomulka**
this commit enables (temporarely) the test to gather some additional logging 
relates https://github.com/elastic/elasticsearch/issues/103286

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by stu-elastic** [comment]
Where's the logging?

**Comment by pgomulka** [comment]
apologies, what I meant was to change the assertion that will log upon failure. the `        assertThat(sampleAssertions, Matchers.anEmptyMap());` is internally only compering the size - expecting 0. we want to see what elements are remaining

File: test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/MetricsApmIT.java [hunk 1]
```diff
@@ -109,7 +108,7 @@ public void testApmIntegration() throws Exception {
         client().performRequest(new Request("GET", "/_use_apm_metrics"));
 
         finished.await(30, TimeUnit.SECONDS);
```
Comment (mosche) [inline test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/MetricsApmIT.java:110]: I suspect the problem is already here, I'd suggest to make sure this returns true.
```suggestion
        assertTrue("Timeout when waiting for assertions to complete.", finished.await(30, TimeUnit.SECONDS));
```

**Review by mosche — APPROVED** [review]
Lgtm :shipit:

**Pull request merged**